File: algorithm/src/test/java/programmers/level1/스마트폰키패드/SolutionTest.java
package programmers.level1.스마트폰키패드;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;


public class SolutionTest {

    @Test
    void name() {
        Solution solution = new Solution();
        // tc1
        String answer = solution.solution(new int[]{1, 3, 4, 5, 8, 2, 1, 4, 5, 9, 5}, "right");
        assertThat(answer).isEqualTo("LRLLLRLLRRL");

        // tc2
        answer = solution.solution(new int[]{7, 0, 8, 2, 8, 3, 1, 5, 7, 6, 2}, "left");
        assertThat(answer).isEqualTo("LRLLRRLLLRR");

        // tc3
        answer = solution.solution(new int[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 0}, "right");
        assertThat(answer).isEqualTo("LLRLLRLLRL");
    }
}

File: algorithm/src/test/kotlin/io/yomni/programmers/level1/id12912/SolutionTest.kt
package io.yomni.programmers.level1.id12912

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource

@DisplayName("SolutionTest")
class SolutionTest {
    companion object {
        @JvmStatic
        fun argumentsProvider() = listOf(
            Arguments.of(3, 5, 12L),
            Arguments.of(3, 3, 3L),
            Arguments.of(5, 3, 12L),
        )
    }

    @ParameterizedTest
    @MethodSource("argumentsProvider")
    fun solution(a: Int, b: Int, expected: Long) {
        val solution = Solution()
        assertThat(solution.solution(a, b)).isEqualTo(expected)
    }
}
